test/formal/combine_byte_shr_1.py

```diff
@@ -0,0 +1,30 @@
+from rule import Rule
+from opcodes import *
+
+"""
+byte(A, shr(B, X))
+given B % 8 == 0 && A <= n_bits/8 && B <= n_bits && A >= B / 8
+->
+byte(A - B / 8, X)
+"""
+
+rule = Rule()
+
+n_bits = 256
+
+# Input vars
+X = BitVec('X', n_bits)
+A = BitVec('A', n_bits)
+B = BitVec('B', n_bits)
+
+# Non optimized result
+nonopt = BYTE(A, SHR(B, X))
+# Optimized result
+opt = BYTE(A - B / 8, X)
+
+rule.require(B % 8 == 0)
+rule.require(ULT(A, n_bits/8))
```

Comment:
```suggestion
rule.require(ULE(A, n_bits/8))
```
Also, do we need to make `n_bits/8` (and `n_bits` below) `BitVecVal`s? Or do literals work?

Reply:
Turns out here and in the other ``ULT``is required and ``A == n_bits/8`` breaks. Not casting to BitVecVal seems fine, though.